src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/TagBuilder.cs

```diff
@@ -25,15 +27,25 @@ public TagBuilder(string tagName)
             }
 
             TagName = tagName;
-            Attributes = new SortedDictionary<string, string>(StringComparer.OrdinalIgnoreCase);
-
             InnerHtml = new BufferedHtmlContent();
         }
 
         /// <summary>
         /// Gets the set of attributes that will be written to the tag.
-        /// </summary>
-        public IDictionary<string, string> Attributes { get; }
+        /// </summary>B
+        public AttributeDictionary Attributes
```

Comment:
:-1: No, the intent is that it should be a concrete type that returns a `struct` enumerator. It's a big problem with using dictionaries in general and there are three ways to squash it.
- don't ever use the enumerator (not likely)
- only use the enumerator from places where you know the concrete type
- do this

All of the arguments about why you should prefer `IDictionary<>` over `Dictionary<>` can also be made for defining your own type. You guys just hate custom collections :trollface: